packages/api/exec.mts

```diff
@@ -135,38 +157,34 @@ export function tsx(options: NodeRequestType) {
  * Install a specific package:
  *
  *     npmInstall({
- *       cwd: '/Users/ben/.srcbook/foo',
- *       package: 'marked',
+ *       cwd: '/path/to/project',
+ *       packages: ['lodash'],
  *       stdout(data) {console.log(data.toString('utf8'))},
  *       stderr(data) {console.error(data.toString('utf8'))},
  *       onExit(code) {console.log(`Exit code: ${code}`)}
  *     });
- *
  */
-export function npmInstall(options: NPMInstallRequestType) {
-  const { cwd, stdout, stderr, onExit } = options;
-  const args = options.packages
-    ? ['install', '--include=dev', ...(options.args || []), ...options.packages]
-    : ['install', '--include=dev', ...(options.args || [])];
-
-  return spawnCall({
+export const npmInstall = ({ cwd, packages = [], args = [], ...rest }: NPMInstallRequestType) =>
+  spawnCall({
     command: 'npm',
     cwd,
-    args,
-    stdout,
-    stderr,
-    onExit,
+    args: ['install', '--include=dev', ...args, ...packages],
     env: process.env,
+    ...rest,
   });
-}
 
 /**
  * Run vite.
  */
-export function vite(options: NpxRequestType) {
-  return spawnCall({
-    ...options,
-    command: Path.join(options.cwd, 'node_modules', '.bin', 'vite'),
-    env: process.env,
+export const vite = ({ cwd, args = [], env = process.env, ...rest }: NpxRequestType) =>
+  spawnCall({
+    command:
+      process.platform === 'win32' ? 'npx.cmd' : Path.join(cwd, 'node_modules', '.bin', 'vite'),
+    cwd,
+    args: process.platform === 'win32' ? ['vite', ...args] : args,
```

Comment:
So on windows, we need to call the command and then add `vite` as an arg? Feels like this is `vite vite`. Surprising

Reply:
Have to hard code it, only then it's running otherwise vite is not getting started, atleast for me it havn't

Reply:
Alien, Yes! While Vite code which was already there didn't started vite and thrown error, require to hard code this in the following way